src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,93 @@
+package duke;
+
+import java.util.Collections;
+import java.util.HashSet;
+
+import duke.command.AddTask;
+import duke.command.Command;
+import duke.command.DeleteTask;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+import duke.exception.ExceptionType;
+
+/**
+ * Parser class which translates input of user to task command, task description and date
+ */
+public class Parser {
+
+    /**
+     * Decipher command based on the input of the user
+     *
+     * @param input Input provided by the user
+     * @return Command to be executed by the Duke Bot
+     * @throws DukeException If the user gives an empty description (except bye and list commands) or an invalid input
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] commandStr = input.trim().split("\\s+");
+        String taskCommand = commandStr[0];
+        String taskDetails = formatInput(taskCommand, input);
+        Command commandType = null;
+
+        switch (taskCommand) {
```

Comment:
the java coding standard suggests that "The explicit //Fallthrough comment should be included whenever there is a case statement without a break statement.", perhaps you could consider adding it in!